app/controllers/posts_controller.rb

```diff
@@ -1,56 +1,59 @@
+# frozen_string_literal: true
 class PostsController < ApplicationController
 
+  before_action :find_post, only: [:show, :edit, :update, :destroy]
 
   def index
     @posts = Post.all
   end
 
+  def new
+    @post = Post.new
+  end
 
   def create
-
-    post = current_user.posts.new(post_params)
-
-    if post.save 
-      flash.notice = "Posted!"
+    @post = current_user.posts.new(post_params)
+    if @post.save 
+      flash.notice = 'Posted!'
+      redirect_to root_path
     else  
-      flash.alert = "Post should not be empty!"
+      flash.alert = 'Post should not be empty!'
+      @posts = Post.all
+      render 'posts/index'
     end 
-    redirect_to root_path
-    
   end
 
+  def show 
+  end 
 
 
   def edit
-    @post = Post.find_by(params[:id])
   end
 
   def update
-    @post = Post.find_by(params[:id])
-    @post.assign_attributes(post_params)
-    if @post.save
-      flash.notice = "Post update successful!!"
+    if @post.update_attributes(content: params[:post][:content])
+        flash.notice = 'Post update successfull!'
         redirect_to root_path
     else 
         render 'edit'
     end 
-    
   end
 
   def destroy
-    @post = Post.find_by(params[:id])
     @post.destroy
```

Comment:
- this is still failing when you try to delete from post detail

![Screenshot 2019-09-10 at 2 14 01 PM](https://user-images.githubusercontent.com/10922508/64616907-58092580-d3d5-11e9-975c-231feb85bd00.png)